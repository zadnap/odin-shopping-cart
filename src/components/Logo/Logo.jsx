import { Link } from 'react-router-dom';
import styles from './Logo.module.scss';
import logo from '@/assets/logo.png';

function Logo() {
  return (
    <Link className={styles.logo} to="/">
      <img className={styles.logoImage} src={logo} alt="Homepage" />
      <h1 className={styles.title}>Cinemart</h1>
    </Link>
  );
}

export default Logo;
