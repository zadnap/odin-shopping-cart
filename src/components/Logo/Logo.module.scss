@use '@/styles/variables.scss' as *;

.logo {
  display: flex;
  align-items: center;
  gap: 10px;

  .logoImage {
    width: 30px;
  }

  .title {
    text-transform: uppercase;
    font-size: 2rem;
    color: $text-color;
  }
}
