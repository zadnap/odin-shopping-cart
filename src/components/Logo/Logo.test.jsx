import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Logo from './Logo.jsx';

describe('Logo component', () => {
  it('should render logo to match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render logo image and title', () => {
    const { getByAltText, getByText } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    );

    expect(getByAltText('Homepage')).toBeInTheDocument();
    expect(getByText('Cinemart')).toBeInTheDocument();
  });

  it('should link to the homepage', () => {
    const { getByRole } = render(
      <MemoryRouter>
        <Logo />
      </MemoryRouter>
    );
    const link = getByRole('link');

    expect(link).toHaveAttribute('href', '/');
  });
});
