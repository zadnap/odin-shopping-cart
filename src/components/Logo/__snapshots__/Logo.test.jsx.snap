// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Logo component > should render logo to match snapshot 1`] = `
<div>
  <a
    class="_logo_5ff345"
    data-discover="true"
    href="/"
  >
    <img
      alt="Homepage"
      class="_logoImage_5ff345"
      src="/src/assets/logo.png"
    />
    <h1
      class="_title_5ff345"
    >
      Cinemart
    </h1>
  </a>
</div>
`;
